Fix regression on loading jit module from flatbuffer

qihqi opened the pull request:
Summary:
https://fb.workplace.com/groups/pytorch.edge.users/permalink/1287477365455887

Root cause:
Introduced in D44106776. But this loop is wierd because class_dep can grow, so it cannot be replaced with c10::irange.

Test Plan:
Used model at `fbpkg fetch speech.tuna.milan.ondevice.en_us.transducer:6`
Then
`buck run xplat/caffe2/fb/lite_predictor:convert_model -- --model=$HOME/20230320debug/pytorchmodel.pt --output_name=/tmp/ffmodel.ff`

Differential Revision: D44234894

larryliu0820 commented on torch/csrc/jit/serialization/python_print.cpp:
```diff
@@ -1745,7 +1745,9 @@ void jitModuleToPythonCodeAndConstants(
     class_deps.add(class_type);
   }
 
-  for (const auto i : c10::irange(class_deps.size())) {
```
what's wrong with c10::irange?

larryliu0820 replied:
Can we add a unit test?

qihqi replied:
inside of the loop might make class_deps longer (somewhere appends to it). Good point on the unit test let me think

qihqi committed:
Fix regression on loading jit module from flatbuffer (#97190)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/97190

https://fb.workplace.com/groups/pytorch.edge.users/permalink/1287477365455887

Root cause:
Introduced in D44106776. But this loop is wierd because class_dep can grow, so it cannot be replaced with c10::irange.

Pickle's version is fine because it is not using irange:
https://www.internalfb.com/code/fbsource/[e4655b053ea9f4e4886492ec7f9c63d05c4138b1]/fbcode/caffe2/torch/csrc/jit/serialization/export_module.cpp?lines=617

Test Plan:
Used model at `fbpkg fetch speech.tuna.milan.ondevice.en_us.transducer:6`
Then
`buck run xplat/caffe2/fb/lite_predictor:convert_model -- --model=$HOME/20230320debug/pytorchmodel.pt --output_name=/tmp/ffmodel.ff`

Reviewed By: Rittzz

Differential Revision: D44234894

fbshipit-source-id: d2eae54eb02bb8f044d29daa392626417601d3b9
